src/deprecated.jl

```diff
@@ -8,3 +8,102 @@ by(args...; kwargs...) = throw(ArgumentError("by function was removed from DataF
 
 aggregate(args...; kwargs...) = throw(ArgumentError("aggregate function was removed from DataFrames.jl. " *
                                                     "Use the `combine` function instead."))
+
+export categorical, categorical!
+function CategoricalArrays.categorical(df::AbstractDataFrame,
+                                       cols::Union{ColumnIndex, MultiColumnIndex};
+                                       compress::Union{Bool, Nothing}=nothing)
+    if compress === nothing
+        compress = false
+        categoricalstr = "categorical"
+    else
+        categoricalstr = "(x -> categorical(x, compress=$compress))"
+    end
+    if cols isa AbstractVector{<:Union{AbstractString, Symbol}}
+        Base.depwarn("`categorical(df, cols)` is deprecated. " *
+                     "Use `transform(df, cols .=> $categoricalstr .=> cols)` instead.",
+                     :categorical)
+        return transform(df, cols .=> (x -> categorical(x, compress=compress)) .=> cols)
+    elseif cols isa Union{AbstractString, Symbol}
+        Base.depwarn("`categorical(df, cols)` is deprecated. " *
+                     "Use `transform(df, cols => $categoricalstr => cols)` instead.",
+                     :categorical)
+        return transform(df, cols => (x -> categorical(x, compress=compress)) => cols)
+    else
+        Base.depwarn("`categorical(df, cols)` is deprecated. " *
+                     "Use `transform(df, names(df, cols) .=> $categoricalstr .=> names(df, cols))` instead.",
+                     :categorical)
+        return transform(df, names(df, cols) .=> (x -> categorical(x, compress=compress)) .=> names(df, cols))
+    end
+end
+
+function CategoricalArrays.categorical(df::AbstractDataFrame,
+                                       cols::Union{Type, Nothing}=nothing;
+                                       compress::Bool=false)
+    if compress === nothing
+        compress = false
+        categoricalstr = "categorical"
+    else
+        categoricalstr = "categorical(x, compress=$compress)"
+    end
+    if cols === nothing
+        cols = Union{AbstractString, Missing}
+        Base.depwarn("`categorical(df, cols)` is deprecated. " *
+                     "Use `cols = names(df)[map(c -> eltype(c) <: $cols, eachcol(df))]; transform(df, cols .=> $categoricalstr .=> cols)` instead.",
+                     :categorical)
+    else
+        Base.depwarn("`categorical(df, cols)` is deprecated. " *
+                     "Use `cols = names(df)[map(c -> eltype(c) <: cols, eachcol(df))]; transform(df, cols .=> $categoricalstr .=> cols)` instead.",
```

Comment:
`cols` variable is used twice here (overwriting the original `cols` vale) so it is probably better to change the name here.